apps/src/lab2/views/components/InstructionsPanel.tsx

```diff
@@ -0,0 +1,232 @@
+  setPredictResponse: (response: string) => void;
+  predictAnswerLocked: boolean;
+  /** Optional classname for the container */
+  className?: string;
+  offerBrowserTts?: boolean;
+  canShowNextButton: boolean;
+  hasNextLevel: boolean;
+  useSecondaryFinishButton: boolean;
+  onContinueOrFinish: () => void;
+  bottomComponent?: React.ReactNode;
+  noTextAnimation?: boolean;
+}
+
+/**
+ * Renders the instructions panel view, which is used within labs as well as to render a preview
+ * of the instructions panel on the level edit page.
+ */
+const InstructionsPanel: React.FunctionComponent<InstructionsPanelProps> = ({
+  text,
+  message,
+  messageIndex,
+  layout = 'vertical',
+  theme = 'Dark',
+  handleInstructionsTextClick,
+  predictSettings,
+  predictResponse,
+  setPredictResponse,
+  predictAnswerLocked,
+  className,
+  offerBrowserTts,
+  canShowNextButton,
+  hasNextLevel,
+  useSecondaryFinishButton,
+  onContinueOrFinish,
+  bottomComponent,
+  noTextAnimation,
+}) => {
+  const vertical = layout === 'vertical';
+
+  const showSecondaryFinishButton = useSecondaryFinishButton && !hasNextLevel;
+
+  const useMessage =
+    showSecondaryFinishButton &&
+    queryParams('show-secondary-finish-button-question') === 'true'
+      ? commonI18n.finishMessage()
+      : message;
+
+  // The secondary finish button avoids a reappearance animation by not using
+  // the unique index.
+  const useMessageIndex = useSecondaryFinishButton ? undefined : messageIndex;
+
+  const feedbackRef = useRef<HTMLDivElement>(null);
+  const runButton = document.querySelector('#run-button');
+
+  useEffect(() => {
+    const checkRunButton = () => {
+      if (feedbackRef.current && runButton?.textContent === 'Run') {
+        feedbackRef.current.focus();
+        return true;
+      }
+      return false;
+    };
+
+    const observer = new MutationObserver(mutations => {
+      if (checkRunButton()) {
+        // Stop observing once the run button is found and has the text 'Run'
+        observer.disconnect();
+      }
+    });
+
+    observer.observe(document.body, {
+      childList: true,
+      subtree: true,
+    });
+
+    // Cleanup observer on component unmount
+    return () => {
+      observer.disconnect();
+    };
+  }, [useMessage, canShowNextButton, runButton?.textContent]);
+
+  return (
+    <div
+      id="instructions"
+      data-theme="Light"
+      className={classNames(
+        moduleStyles['instructions-' + theme],
+        vertical && moduleStyles.vertical,
+        'instructions',
+        className
+      )}
+    >
+      <div
+        id="instructions-panel"
+        className={classNames(
+          moduleStyles.item,
+          vertical && moduleStyles.itemVertical
+        )}
+      >
+        {text && (
+          <div
+            key={text}
+            id="instructions-text"
+            className={classNames(
+              moduleStyles['text-' + theme],
+              noTextAnimation && moduleStyles.noAnimation
```

Comment:
The vast vast majority of changes here are just copying and pasting the `InstructionsPanel` component from the `Instructions` file into its own file.

One small change was to add this `noTextAnimation` prop so that we don't apply an animation on each rerender in the preview.